### PR TITLE
INGM-447 Set the boot_in_app flag according to the file extension

### PR DESCRIPTION
**Pull request by pzinemanas**
### Description

This PR sets the boot_in_app flag according to the file extension

Fixes # INGM-447

### Type of change

- [x] Sets the boot_in_app flag according to the file extension
- [x] Change 2 (description)

### Tests
- [x] Test the load FW using the unreleased FoE application.
- [x] Run tests.

Please describe the tests that you ran to verify your changes if it applies. 

### Documentation

Please update the documentation.

- [x] Update docstrings of every function, method or class that change.
- [x] USe [type hints](https://docs.python.org/3/library/typing.html) for every function and argument.
- [x] Build documentation locally to verify changes.
- [x] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting

- [x] Use black package to format the code: `black -l 100 ingeniamotion tests`. It is recommended to configure the code editor to automatically format the code using black with a max length line of 100.

### Others

- [x] Set fix version field in the Jira issue.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -7,7 +7,7 @@ env_list = coverage, build, firmware, docs, format, type, py{39,310,311,312}, vi
 # For instance, to install ingenialink from a custom branch.
 # It should be empty in production.
 [develop]
-    ingenialink = git+https://github.com/ingeniamc/ingenialink-python@develop
+    ingenialink = git+https://github.com/ingeniamc/ingenialink-python@INGK-892-add-boot-in-app-flag-in-the-load-firmware-of-ethercat
```
Comment (pzinemanas) [inline tox.ini:10]: Undo this change when https://github.com/ingeniamc/ingenialink-python/pull/414 is merged

File: ingeniamotion/communication.py [hunk 2]
```diff
@@ -999,6 +1001,25 @@ def load_firmware_canopen(
                 error_enabled_callback,
             )
 
+    @staticmethod
+    def __get_boot_in_app(fw_file: str) -> bool:
+        """Return true if the FoE is included in the application (file extension is .sfu)
```
Comment (MartinALL91) [inline ingeniamotion/communication.py:1006]: If the bootloader is included in the application.

File: ingeniamotion/communication.py [hunk 3]
```diff
@@ -1253,13 +1282,23 @@ def __load_ensemble_fw_ecat(self, net: EthercatNetwork, fw_file: str, slave: int
             net: Ethercat network.
             fw_file: Path to the ensemble FW file.
             slave: Slave ID (any slave in the ensemble)
+            boot_in_app: true if the bootloader is included in the application, false otherwise.
```
Comment (GullinBustin) [inline ingeniamotion/communication.py:1285]: You can explain what happens when it is None

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: ingeniamotion/communication.py (hunk 3 above)
Comment (pzinemanas) [reply ingeniamotion/communication.py:1285]: Done!

**Review by MartinALL91 — APPROVED** [review]

**Review by GullinBustin — APPROVED** [review]

File: ingeniamotion/communication.py [hunk 4]
```diff
@@ -999,17 +1001,41 @@ def load_firmware_canopen(
                 error_enabled_callback,
             )
 
-    def load_firmware_ecat(self, ifname: str, fw_file: str, slave: int = 1) -> None:
+    @staticmethod
+    def __get_boot_in_app(fw_file: str) -> bool:
+        """Return true if the bootloader is included in the application (file extension is .sfu).
```
Comment (polfeliu) [inline ingeniamotion/communication.py:1006]: This is not obvious, I would include this on the description

CoCo-MoCo products have boot_in_app, and exclusively use .sfu fw file formats.
CoMoCo products do not have boot in app, and exclusively use .lfu fw file formats
